### PR TITLE
Unpin `camera_android` and remove its only usage

### LINKED ISSUE #146004: Unpin camera_android
camera_android is pinned because of failures in #145959.

Pinning PR: https://github.com/flutter/flutter/pull/145959

### PR DESCRIPTION
**Pull request by gmackall**
Fixes https://github.com/flutter/flutter/issues/146004

The `camera_android`plugin had its version pinned when it was still used in the `gradle_deprecated_settings`. It has since been [replaced](https://github.com/flutter/flutter/pull/148426) by `camera_android_camerax`, and therefore isn't used in `flutter/flutter`. So it's pin should be fine to remove.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [ ] All existing and new tests are passing.

### CODE REVIEW

**Review by reidbaker — APPROVED** [review]

File: dev/integration_tests/gradle_deprecated_settings/android/project-camera_android.lockfile [hunk 1]
```diff
@@ -1,157 +0,0 @@
-# This is a Gradle generated file for dependency locking.
-# Manual edits can break the build and are not advised.
```
Comment (christopherfujino) [inline dev/integration_tests/gradle_deprecated_settings/android/project-camera_android.lockfile:2]: Do we not need a lockfile for this test app?

Comment (gmackall) [reply dev/integration_tests/gradle_deprecated_settings/android/project-camera_android.lockfile:2]: This lockfile was replaced by the one for `camera_android_camerax`
https://github.com/flutter/flutter/blob/master/dev/integration_tests/gradle_deprecated_settings/android/project-camera_android_camerax.lockfile

Comment (christopherfujino) [reply dev/integration_tests/gradle_deprecated_settings/android/project-camera_android.lockfile:2]: ahh, this is per dep, thanks!

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**